### PR TITLE
Validate cluster and rule ids

### LINKED ISSUE #295: Implement validation of cluster_id for like, dislike and rest_vote endpoints
Currently, any authenticated user can vote on any non-existing cluster. We should prevent this behavior. If a user tries to access to non-existsing cluster 404 should be returned.

### LINKED ISSUE #296: Implement validation of rule_id for like, dislike and reset_vote endpoints.
Should be started after finishing #280. The use case is similar to #295.

### LINKED ISSUE #398: Implement a check of foreign keys in testing env
now all foreign key errors are ignored

### PR DESCRIPTION
**Pull request by Sergey1011010**
# Description

Fixes #295
Fixes #296
Fixes #398

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update
- Unit tests (no changes in the code)

## Testing steps

`make before_commit`

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -248,6 +256,42 @@ func (server *HTTPServer) voteOnRule(writer http.ResponseWriter, request *http.R
...
+		return false
+	}
+	if !clusterExists {
+		responses.Send(
+			http.StatusNotFound,
+			writer,
+			(&storage.ItemNotFoundError{ItemID: clusterID}).Error(),
+		)
+		return false
+	}
```
Comment (Bee-lee) [inline server/server.go]: Why not return `storage.ItemNotFoundError` from storage and let `handleServerError` handle it? Or do you not like that approach?

File: server/server.go [hunk 2]
```diff
@@ -248,6 +256,42 @@ func (server *HTTPServer) voteOnRule(writer http.ResponseWriter, request *http.R
...
+
+	ruleExists, err := server.Storage.CheckIfRuleExists(ruleID)
+	if err != nil {
+		log.Err(err)
```
Comment (Bee-lee) [inline server/server.go]: Shouldn't it be `log.Error().Err(err)` ? Is it the same, just shorter?

File: server/server.go [hunk 3]
```diff
@@ -248,6 +256,42 @@ func (server *HTTPServer) voteOnRule(writer http.ResponseWriter, request *http.R
...
+		responses.Send(http.StatusInternalServerError, writer, err.Error())
+		return false
+	}
+	if !ruleExists {
+		responses.Send(
```
Comment (Bee-lee) [inline server/server.go]: the returned err from responses.Send* is not handled

File: server/server.go (hunk 1 above)
Comment (Sergey1011010) [reply server/server.go]: hm, maybe actually better, thanks

File: server/server.go (hunk 2 above)
Comment (Sergey1011010) [reply server/server.go]: not sure

File: server/server.go (hunk 3 above)
Comment (Sergey1011010) [reply server/server.go]: it's gonna be huge refactoring since we're not checking it everywhere :)

File: server/server.go (hunk 2 above)
Comment (Sergey1011010) [reply server/server.go]: fixed

File: server/server.go (hunk 3 above)
Comment (Bee-lee) [reply server/server.go]: we are, it was in my PR yesterday. That's also why the `handleServerError` function, so it doesn't have to be everywhere :D

File: server/server.go (hunk 1 above)
Comment (Sergey1011010) [reply server/server.go]: where is this `handleServerError`?

Comment (Bee-lee) [reply server/server.go]: i guess you dont have master merged

Comment (Bee-lee) [reply server/server.go]: server/errors.go

Comment (Sergey1011010) [reply server/server.go]: nevermind, I'm using an old version of base branch

File: server/server.go (hunk 3 above)
Comment (tisnik) [reply server/server.go]: +1

File: server/server.go (hunk 1 above)
Comment (Sergey1011010) [reply server/server.go]: fixed

File: server/server.go (hunk 3 above)
Comment (Sergey1011010) [reply server/server.go]: fixed

**Pull request merged**